axiom/tokens.go

```diff
@@ -0,0 +1,239 @@
+package axiom
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+//go:generate go run golang.org/x/tools/cmd/stringer -type=Action -linecomment -output=action_string.go
+
+// Action represents an action that can be performed on an Axiom resource.
+type Action uint8
+
+// All available [Action].
+const (
+	ActionCreate Action = iota // create
+	ActionRead                 // read
+	ActionUpdate               // update
+	ActionDelete               // delete
+)
```

Comment:
I think it makes sense to use an `emptyAction` here (instead of `iota + 1`) so `Action(0)` doesn't evaluate to `ActionCreate`.

Check https://github.com/axiomhq/axiom-go/blob/main/axiom/orgs.go#L20-L29. The `orgs_test.go` file also has a pattern for testing a set of constants.